testdata/p4tc_samples_outputs/add_entry_1_example_parser.h

```diff
@@ -65,3 +65,48 @@ REGISTER_TABLE(hdr_md_cpumap, BPF_MAP_TYPE_PERCPU_ARRAY, u32, struct hdr_md, 2)
 BPF_ANNOTATE_KV_PAIR(hdr_md_cpumap, u32, struct hdr_md)
 REGISTER_END()
 
+static inline u32 getPrimitive32(u8 *a, int size) {
+   if(size <=32 || size >=56) {
+       bpf_printk("Invalid size.");
+   };
+   return  ((((u32)a[2]) <<16) | (((u32)a[1]) << 8) | a[0]);
+}
+static inline u64 getPrimitive64(u8 *a, int size) {
+   if(size <=32 || size >=56) {
+       bpf_printk("Invalid size.");
+   };
+   if(size <= 40) {
+       return  ((((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
+   } else {
+       if(size <= 48) {
+           return  ((((u64)a[5]) << 40) | (((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
+       } else {
+           return  ((((u64)a[6]) << 48) | (((u64)a[5]) << 40) | (((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
+       }
+   }
+}
+static inline void storePrimitive32(u8 *a, int size, u32 value) {
+   if(size <=16 || size >=24) {
```

Comment:
It also seems incorrect here
If you call `storePrimitive32` with size = 24, for example, it will print Invalid size